Rain test fails randomly

**Describe the bug**
Rain test fails after 2038-01-19

**To Reproduce**

on openSUSE, I build the package with
```bash
osc co openSUSE:Factory/python-asciimatics
osc build --vm-type=kvm --noservice --clean --build-opt=--vm-custom-opt="-rtc base=2038-01-20T00:00:00" --alternative-project=home:bmwiedemann:reproducible openSUSE_Tumbleweed
```

**Expected behavior**

tests should continue to pass in future

**Screenshots**
```
 =================================== FAILURES ===================================
 ___________________________ TestParticles.test_rain ____________________________
 
 self = <tests.test_particles.TestParticles testMethod=test_rain>
 
     def test_rain(self):
         """     
         Test that Rain works as expected.
         """     
         screen = MagicMock(spec=Screen, colours=8)
         canvas = Canvas(screen, 10, 40, 0, 0)
         effect = Rain(canvas, 200)
 >       self.check_effect(canvas,
                           effect,
                           lambda value: self.assertIn(chr(value[0]), ' `\\v'))

 tests/test_particles.py:98:
 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
 tests/test_particles.py:43: in check_effect
     self.assertTrue(changed, "failed at step %d %s" % (i, view))
 E   AssertionError: False is not true : failed at step 0

==== 1 failed, 149 passed, 41 skipped, 1 deselected, 966 warnings in 3.67s =====
```

**System details (please complete the following information):**
 - OS and version: openSUSE-Tumbleweed-20220906
 - Python version: 3.10
 - Python distribution: cpython
 - Asciimatics version 1.14.0

**Additional context**
See also https://en.wikipedia.org/wiki/Year_2038_problem

As part of my work on reproducible builds for openSUSE, I check that software still gives identical build results in the future.
The usual offset is +16 years, because that is how long I expect some software will be used in some places.
This showed up failing tests in our package build.
See https://reproducible-builds.org/ for why this matters.

--- a/tests/test_particles.py
+++ b/tests/test_particles.py
@@ -1,3 +1,4 @@
+import random
 import unittest
 from mock.mock import MagicMock
 from asciimatics.particles import ShootScreen, DropScreen, Explosion, Rain, \
@@ -6,6 +7,9 @@ from asciimatics.screen import Screen, Canvas
 
 
 class TestParticles(unittest.TestCase):
+    def setUp(self):
+        random.seed(42)
+
     def check_effect(self, canvas, effect, assert_fn,
                      is_blank=True, iterations=40, warm_up=0):
         """

